docs/WASI-tutorial.md

```diff
@@ -244,3 +249,59 @@ The capability model is very powerful, and what's shown here is just the beginni
 In the future, we'll be exposing much more functionality, including finer-grained
 capabilities, capabilities for network ports, and the ability for applications to
 explicitly request capabilities.
+
+## Web assembly text example
+
+In this example we will look at compiling the WebAssembly text format into wasm, and
+running the compiled WebAssembly module using the `wasmtime` runtime. This example
+makes use of WASI's `fd_write` implementation to write `hello world` to stdout.
+
+First, create a new `demo.wat` file:
+
+```wat
+(module
+    ;; Import the required fd_write WASI function which will write the given io vectors to stdout
+    ;; (File Descriptor, *iovs, iovs_len, nwritten) -> Returns number of bytes written
+    (import "wasi_unstable" "fd_write" (func $fd_write (param i32 i32 i32 i32) (result i32)))
+
+    (memory 1)
+    (export "memory" (memory 0))
+
+    ;; Write 'hello world\n' to memory at an offset of 8 bytes
+    ;; Note the trailing newline which is required for the text to appear
+    (data (i32.const 8) "hello word\n")
```

Comment:
```suggestion
    (data (i32.const 8) "hello world\n")
```

Reply:
Looks great to me! Thanks! One thing though, did you know that you don't need to compile WAT to WASM in order to run in wasmtime? You could actually simply run:

```
$ wasmtime demo.wat
```

Perhaps it'd be good to add in the tutorial as well, next to

```
$ wat2wasm demo.wat
```

What do you think?